src/interpreter/ByteCodeInterpreter.cpp

```diff
@@ -55,11 +55,15 @@ ALWAYS_INLINE size_t resolveProgramCounter(char* codeBuffer, const size_t progra
     return programCounter - (size_t)codeBuffer;
 }
 
-Value ByteCodeInterpreter::interpret(ExecutionState& state, ByteCodeBlock* byteCodeBlock, size_t programCounter, Value* registerFile, void* initAddressFiller)
+Value ByteCodeInterpreter::interpret(ExecutionState& state, ByteCodeBlock* byteCodeBlock, size_t programCounter, Value* registerFile)
 {
 #if defined(COMPILER_GCC)
-    if (UNLIKELY(initAddressFiller != nullptr)) {
-        *((size_t*)initAddressFiller) = ((size_t) && FillOpcodeTableOpcodeLbl);
+    // Initialize the OpcodeTable.
+    if (UNLIKELY(byteCodeBlock == nullptr)) {
+#define REGISTER_TABLE(opcode, pushCount, popCount) g_opcodeTable.m_table[opcode##Opcode] = &&opcode##OpcodeLbl;
+        FOR_EACH_BYTECODE_OP(REGISTER_TABLE);
+#undef REGISTER_TABLE
```

Comment:
Locating a large code part for `OpcodeTable` initialization at the upper most part in interpreter may negatively affect the code cache. How about moving this initialization code into `OpcodeTable::OpcodeTable()` ?

Reply:
The `OpcodeTable` initialization must be in `ByteCodeInterpreter::interpret` because the addresses of the opcode labels only accessible in that scope.
I've measured the performance (release, x86, gcc) and the results are the same if I put the OpcodeTable initialization to the upper part of `ByteCodeInterpreter::interpret` or to the end of the function.

|Testname|First commit runtime(6d7312579ee9ce9f114e8551cb6e3c76697862e6)|Second commit runtime(ac055b8731ca57658a549ad4ab0dfb9afdd71475)|Gain
|-|-|-|-|
|3d-cube|36.5 ms (± 0.41 ms)|36.5 ms (± 0.55 ms)|-0.00% (+0.00 ms) *(noise)*|
|3d-morph|33.5 ms (± 0.41 ms)|33.3 ms (± 0.37 ms)|+0.60% (-0.20 ms) *(noise)*|
|**3d-raytrace**|43.6 ms (± 0.40 ms)|43 ms (± 0.00 ms)|**+1.38%** (-0.60 ms) |
|**access-binary-trees**|25.5 ms (± 0.41 ms)|25 ms (± 0.00 ms)|**+1.96%** (-0.50 ms) |
|access-fannkuch|70.3 ms (± 0.37 ms)|70 ms (± 0.00 ms)|+0.43% (-0.30 ms) *(noise)*|
|access-nbody|40.3 ms (± 0.37 ms)|40.2 ms (± 0.33 ms)|+0.25% (-0.10 ms) *(noise)*|
|access-nsieve|13.8 ms (± 0.33 ms)|13.8 ms (± 0.33 ms)|-0.00% (+0.00 ms) *(noise)*|
|bitops-3bit-bits-in-byte|12 ms (± 0.00 ms)|12 ms (± 0.00 ms)|-0.00% (+0.00 ms) *(noise)*|
|bitops-bits-in-byte|20.3 ms (± 0.37 ms)|20.1 ms (± 0.24 ms)|+0.99% (-0.20 ms) *(noise)*|
|bitops-bitwise-and|21 ms (± 0.00 ms)|21 ms (± 0.00 ms)|-0.00% (+0.00 ms) *(noise)*|
|bitops-nsieve-bits|25 ms (± 0.00 ms)|25 ms (± 0.00 ms)|-0.00% (+0.00 ms) *(noise)*|
|controlflow-recursive|12 ms (± 0.00 ms)|12 ms (± 0.00 ms)|-0.00% (+0.00 ms) *(noise)*|
|crypto-aes|31 ms (± 0.00 ms)|31 ms (± 0.00 ms)|-0.00% (+0.00 ms) *(noise)*|
|crypto-md5|13.9 ms (± 0.24 ms)|14 ms (± 0.00 ms)|-0.72% (+0.10 ms) *(noise)*|
|crypto-sha1|14.8 ms (± 0.33 ms)|14.8 ms (± 0.33 ms)|-0.00% (+0.00 ms) *(noise)*|
|date-format-tofte|87.6 ms (± 0.40 ms)|87.7 ms (± 0.37 ms)|-0.11% (+0.10 ms) *(noise)*|
|date-format-xparb|66.5 ms (± 0.98 ms)|67.3 ms (± 0.64 ms)|-1.20% (+0.80 ms) *(noise)*|
|math-cordic|32 ms (± 0.00 ms)|32 ms (± 0.00 ms)|-0.00% (+0.00 ms) *(noise)*|
|math-partial-sums|22.8 ms (± 0.33 ms)|22.9 ms (± 0.24 ms)|-0.44% (+0.10 ms) *(noise)*|
|math-spectral-norm|17 ms (± 0.00 ms)|16.6 ms (± 0.40 ms)|+2.35% (-0.40 ms) *(noise)*|
|regexp-dna|126 ms (± 0.00 ms)|126.1 ms (± 0.24 ms)|-0.08% (+0.10 ms) *(noise)*|
|string-base64|39 ms (± 0.00 ms)|39 ms (± 0.00 ms)|-0.00% (+0.00 ms) *(noise)*|
|string-fasta|60.6 ms (± 0.40 ms)|60.5 ms (± 0.41 ms)|+0.17% (-0.10 ms) *(noise)*|
|**string-tagcloud**|63.6 ms (± 0.40 ms)|64.1 ms (± 0.44 ms)|**-0.79%** (+0.50 ms) |
|string-unpack-code|97 ms (± 0.52 ms)|97 ms (± 0.63 ms)|-0.00% (+0.00 ms) *(noise)*|
|string-validate-input|36.4 ms (± 0.40 ms)|36.8 ms (± 0.33 ms)|-1.10% (+0.40 ms) *(noise)*|